lib/ex_unit/lib/ex_unit/assertions.ex

```diff
@@ -359,9 +359,14 @@ defmodule ExUnit.Assertions do
         "\n  " <> unquote(binary) <> " = " <> inspect(unquote(var))
       end
     end)
+    |> Enum.reduce(fn(pin, acc) ->
+      quote do
+        unquote(acc) <> unquote(pin)
+      end
+    end)
 
     quote do
-      "\nThe following variables were pinned:" <> unquote_splicing(content)
+      "\nThe following variables were pinned:" <> unquote(content)
```

Comment:
It'd be better to keep `unquote_splicing/1` with:
```elixir
<<"\nThe following variables were pinned:", unquote_splicing(content)>>
```
and no need for extra `Enum.reduce/2` then.